tempodb/tempodb.go

```diff
@@ -544,3 +546,28 @@ func (rw *readerWriter) compactedBlocklist(tenantID string) []*encoding.Compacte
 
 	return copiedBlocklist
 }
+
+func (rw *readerWriter) cleanMissingTenants(tenants []string) {
```

Comment:
can we write some tests that target this method directly?  use a slice of test cases pattern.